### PR TITLE
Added `serde` support for `RowGroupMetaData`.

### PR DESCRIPTION
**Pull request by youngsofun**
use thrift for types in parquet_format_safe.

https://github.com/jorgecarleitao/parquet2/issues/200

it turns out that there are as many as 16 types that need to impl `De/Serialize`, so it is very troublesome to work it around outside.

### CODE REVIEW

**Comment by jorgecarleitao** [comment]
This would add a significant dependency to this crate. Would it make sense to make this an optional feature under `serde`. Something like this: https://github.com/jorgecarleitao/arrow2/blob/main/src/datatypes/field.rs#L3

**Comment by youngsofun** [comment]
@jorgecarleitao refactor done, it is a feature now.

File: src/metadata/column_chunk_metadata.rs [hunk 1]
```diff
@@ -7,16 +9,63 @@ use crate::compression::Compression;
 use crate::error::{Error, Result};
 use crate::schema::types::PhysicalType;
 use crate::statistics::{deserialize_statistics, Statistics};
+#[cfg(feature = "serde_types")]
```
Comment (youngsofun) [inline src/metadata/column_chunk_metadata.rs:12]: do you feel better if we group them together?

```
#[cfg(feature = "serde_types")]
mod serde_types {
    pub use parquet_format_safe::thrift::protocol::{
        TCompactInputProtocol, TCompactOutputProtocol,
    };
    pub use serde::de::Error as DeserializeError;
    pub use serde::ser::Error as SerializeError;
    pub use serde::{Deserialize, Deserializer, Serializer};
    pub use serde_derive::{Deserialize, Serialize};
    pub use std::io::Cursor;
}

#[cfg(feature = "serde_types")]
use serde_types::*;
```

**Comment by youngsofun** [comment]
@jorgecarleitao  Is there anything else needs to improve?

**Review by sundy-li — APPROVED** [review]

**Comment by jorgecarleitao** [comment]
Awesome, thanks a lot!

**Comment by youngsofun** [comment]
Is it ready to bump to 0.16.4? @jorgecarleitao

**Comment by jorgecarleitao** [comment]
We need to bump to 0.17 - there are breaking changes in main. Released :)

**Pull request merged**